### ISSUE TITLE
[Feature Request] Generate numeric enums

### ISSUE THREAD

**Issue by pablopaz** [odata2ts/odata2ts#271]
odata2ts version: 0.31.1

Hi, i'm new to both odata and this library.

In our project, we use a DB First approach and we have a bunch of enums that map to ids in the DB. I wanted to manage the enums in one place and have odata2ts generate them so I can use them in the front.

But it seems that the generated enums are always string enums. It would be useful for my use case to have them generated with the numeric value that maps to de DB Id.

For this EnumTypes in my metadata
```
<Schema>
  <EnumType Name="Status">
    <Member Name="Pending" Value="1"/>
    <Member Name="Enqueued" Value="2"/>
    <Member Name="Sended" Value="3"/>
    <Member Name="Failed" Value="4"/>
    <Member Name="Canceled" Value="5"/>
  </EnumType>
  <EnumType Name="ContactType">
    <Member Name="VIDEO" Value="1"/>
    <Member Name="PHONE" Value="3"/>
    <Member Name="EMAIL" Value="7"/>
  </EnumType>
</Schema>
```
the generated enums are as follows:
```
export declare enum Status {
    Pending = "Pending",
    Enqueued = "Enqueued",
    Sended = "Sended",
    Failed = "Failed",
    Canceled = "Canceled"
}
export declare enum ContactType {
    VIDEO = "VIDEO",
    PHONE = "PHONE",
    EMAIL = "EMAIL"
}
```

The expected for my need would be:
```
export declare enum ContactType {
    VIDEO = 1,
    PHONE = 3,
    EMAIL = 7
}
```

Thanks for your time.

**Comment by texttechne** [comment]
Hi @pablopaz,
thanks for bringing that up, I think I neglected enums for a while now... 
But you're right on track: Numeric enums are much more in the spirit of OData enums.

The problem for me there was, that the `UnderlyingType` prop may be any integer and thus also `Edm.Int64`. However, this might be larger than a `number` and could be represented by `bigint`, which then wouldn't work with numeric enums... 

After some consideration, I think that using `Edm.Int64` is an edge case which can be neglected or, respectively, handled when someone brings up a good use case...

**Comment by texttechne** [comment]
Hi @pablopaz,

so I've implemented that (not released yet), but there's a show stopper:
```ts
const model: EditablePersonModel = {
      user: "williams",
      firstName: "Heinz",
      // favoriteFeature: FeatureModel.Feature1, // <= here is how it's used today
     favoriteFeature: FeatureModel[FeatureModel.Feature1] // users would need to do that at every corner
};
```

The thing here is, that in OData you send and get back strings for enums, not numbers. 
Also, within the query you need to use a string:
```
GET {{baseUrl}}/People?$select=FirstName,Gender&$filter=Gender eq 'Female'
```

In short, while the representation as numeric enum would be more in alignment with the OData spec, it's utterly unpractical to use numeric enums.

**Comment by texttechne** [comment]
Please explain your use case a bit more. You are only interested in the models, not in the full OData client, right?

**Comment by pablopaz** [comment]
Hi,
it makes sense that OData uses the name to send and get back the enums, and i understand it may be unpractical.

As for our use case, in fact we are not using enums in the queries, as the models generated have number type.

```
const model: EditableUser = {
      LastName: "williams",
      FirstName: "Heinz",
      UserTypeId: UserTypeEnum.Admin // The field is not an enum in the OData model, it would be a number.
};
```
There would be a UserType table in the DB, and UserTypeEnum.Admin would match the appropriate id in that table.

And we do the queries comparing the id:
```
GET {{baseUrl}}/People?$select=FirstName,UserTypeId&$filter=UserTypeId eq 3
```

In this case, without changing the EF model generation, the only other solution i can think of is to have the enums definition duplicated, both in the Back-End and the Front-End. I thought that maybe we could take advantage of odata2ts generation to manage the enums mapping to the ids in only one place.

We are more interested in the models, as we are not using the full client right now, but may consider using it in the future.

**Comment by texttechne** [comment]
Thanks for this insight. Now it makes more sense.
But to be sure, you're making no use at all of enums within your data model, meaning that neither any EntityType nor any ComplexType is referencing these enums directly? Or do you have multiple use cases?
If you're using enums differently, then it will become more complicated..

In general I see two approaches here: 
1. Introduce an extra config option to generate numeric enums
2. Generating numeric enums and introducing a proper enum converter which works with strings instead

But actually I have to ponder more about this...

**Comment by pablopaz** [comment]
> But to be sure, you're making no use at all of enums within your data model, meaning that neither any EntityType nor any ComplexType is referencing these enums directly?

Correct, the enums are not referenced in the model.

**Comment by texttechne** [comment]
For me later: [TS implementation hint for QEnumPath](https://www.typescriptlang.org/play/?#code/KYOwrgtgBAsgngUXNA3gKCpqBBKBeKAIm0IBoMsAhfIysizAYRsMcLQF800AXOAB2BQAyjwCGIACZiATpKSQAPABUAfDXRYoAbQCWkgFxQAzjxm6QAcwC6R5VAA+JsxcsBuBjvBHkAI2Ayts7mVh5caADGADZixsZQAOKgAboRjDFxKlDAAB48oJLxohLScgoQimAgANYgAPYA7iCq6ppQEXUgpjJgETx1MgAU-GC+UalQEBbAdgCUUCjhmCNjE8AAjmBiUcaDoJAAattgM1DK2tXAcHUAZmfW822YAPTPUAACPMYAtLqW9TJTv06lAbhIInBSFAGsAAOSAyZifj8VxQXxiCLVKD9bEACyE3VRADdjsBPB0ujxgq4jlETjQeLjdMYAHRTEDAbT7CC0k7WDxaQE8MAyEBQAAGOWy6ygsIAJChCVZecAOLDxRQuOEKaYoFKCByGolkuY0hljIo+IJbrBEMhVIN4OVZh5Ip1jHUosAWVE6pZBoQAErAYxgKI8IxkPUsjZbHaOu2QFnYWazIA)

**Comment by texttechne** [comment]
Hi @pablopaz,

so this feature has finally landed with [@odata2ts/odata2ts@0.38.0](https://github.com/odata2ts/odata2ts/releases/tag/%40odata2ts%2Fodata2ts%400.38.0): 
* You need to activate it though: `numericEnums: true` (it's a base option)

Took quite some time to refactor internals to get an improved enum support including numeric enums (that is, for the full odata client). However, all of this work has nothing to do with your use case. Hence, I'm going to close this issue right away. Please reopen if stuff doesn't work.

**Comment by pablopaz** [comment]
Hi @texttechne,

Sorry for the late reply. I just tried it, and it seems to be working fine for my use case.

Thanks!
